src/test/java/io/cdap/plugin/gcp/dataplex/sink/config/DataplexBatchSinkConfigTest.java

```diff
@@ -576,4 +615,729 @@ public void validateFormatForStorageBucketWhenFormatIsNotNull() {
+    assertEquals(" filter", config.getPartitionFilter());
+  }
+
+  @Test
+  public void testGetSchema() {
+    MockFailureCollector failureCollector = new MockFailureCollector();
+    String schema = "Schema.recordOf(\"record\",\n" +
+      "                                    Schema.Field.of(\"id\", Schema.of(Schema.Type.LONG)));";
+    DataplexBatchSinkConfig config = getBuilder().setSchema(schema).build();
+    try {
+      config.getSchema(failureCollector);
+      Assert.fail("Exception is not thrown on valid schema");
+    } catch (ValidationException e) {
+      assertEquals(1, failureCollector.getValidationFailures().size());
+
+    }
+  }
+
+  @Test
+  public void testValidateOutputFormatProvider() {
+    DataplexBatchSinkConfig.Builder builder = getBuilder();
+    DataplexBatchSinkConfig dataplexBatchSinkConfig = PowerMockito.spy(builder.setTable("test").build());
+    MockFailureCollector failureCollector = new MockFailureCollector();
+    FormatContext context = Mockito.mock(FormatContext.class);
+    String format = "format";
+    ValidatingOutputFormat validatingOutputFormat = Mockito.mock(ValidatingOutputFormat.class);
+    dataplexBatchSinkConfig.validateOutputFormatProvider(context, format, validatingOutputFormat);
+    assertEquals(0, failureCollector.getValidationFailures().size());
+  }
+
+  @Test
+  public void testValidateOutputFormatProviderWNull() {
+    DataplexBatchSinkConfig.Builder builder = getBuilder();
+    DataplexBatchSinkConfig dataplexBatchSinkConfig = PowerMockito.spy(builder.setTable("test").build());
+    MockFailureCollector failureCollector = new MockFailureCollector();
+    FormatContext context = Mockito.mock(FormatContext.class);
+    Mockito.when(context.getFailureCollector()).thenReturn(failureCollector);
+    String format = "format";
+    try {
+      dataplexBatchSinkConfig.validateOutputFormatProvider(context, format, null);
+    } catch (ValidationException e) {
+      assertEquals(1, failureCollector.getValidationFailures().size());
+    }
+  }
+
+  @Test
+  public void testValidateBigQueryDataset() throws Exception {
+    io.cdap.cdap.api.data.schema.Schema schema = io.cdap.cdap.api.data.schema.Schema.recordOf("record",
+      io.cdap.cdap.api.data.schema.Schema.Field.of("order",
+        io.cdap.cdap.api.data.schema.Schema.of(io.cdap.cdap.api.data.schema.Schema.Type.LONG)));
+    MockFailureCollector mockFailureCollector = new MockFailureCollector();
+    GCPConnectorConfig dataplexConnectorConfig = mock(GCPConnectorConfig.class);
+    GoogleCredentials googleCredentials = Mockito.mock(GoogleCredentials.class);
+    PowerMockito.mockStatic(DataplexUtil.class);
+    DataplexServiceClient dataplexServiceClient = mock(DataplexServiceClient.class);
+    Mockito.doNothing().when(dataplexServiceClient).close();
+    when(dataplexConnectorConfig.isServiceAccountJson()).thenReturn(new Boolean(Boolean.TRUE));
```

Comment:
`new Boolean(Boolean.TRUE)` can be simplified to just `Boolean.TRUE`

Reply:
done